Test remote sharing

nickvergessen opened the pull request:
- Copied lib/oc-tests/test_shareFiles.py
- Fixed the shares to be remote shares
- Add methods to accept the shares before syncing

Test is currently failing, because deleting a remote shared file does not remove the file on the other end:

```
2015-06-03 17:39:30,269 - ERROR - sharee_two -
File /sharee_two/TEST_FILE_USER_RESHARE.dat exists but should not
2015-06-03 17:39:31,284 - ERROR - sharee_one -
File /sharee_one/TEST_FILE_USER_SHARE.dat exists but should not
```

Requires:
- [x] https://github.com/owncloud/core/pull/15895
- [x] https://github.com/owncloud/core/pull/16733
- [x] https://github.com/owncloud/pyocclient/pull/96

Fails due to:
- [x] https://github.com/owncloud/core/issues/16751

Fix #13

nickvergessen committed:
Add an skeleton for remote share testing

nickvergessen committed:
Correctly build the sharee for remote shares

nickvergessen committed:
Add methods to manage remote shares

nickvergessen committed:
Fix method name

nickvergessen commented on lib/oc-tests/test_remoteShareFile.py:
```diff
@@ -0,0 +1,272 @@
+    user2 = "%s%i" % (config.oc_account_name, 2)
+
+    kwargs = {'perms': sharePermissions, 'remote_user': True}
+    shared['TEST_FILE_USER_SHARE'] = share_file_with_user(
```
check return values to be `> 0`

nickvergessen commented on lib/oc-tests/test_remoteShareFile.py:
```diff
@@ -0,0 +1,272 @@
+
+    # Accept the remote shares for user3
+    user3 = "%s%i" % (config.oc_account_name, 3)
+    openShares = list_remote_share(user3)
```
rename method

nickvergessen committed:
The file should still be synced, but it's not reachable anymore

nickvergessen committed:
Move methods into a new python package

nickvergessen commented:
This should be ready for merge now.

PVince81 commented:
Some errors against stable9 (b3fef3900656b66d938067c6b8c14b1c94efd1a6)

```
SUMMARY:smash.:runid: 160321-173025
SUMMARY:smash.:URL: http://test:***@127.0.0.1/owncloud/remote.php/webdav/
SUMMARY:smash.:Running iteration 1
SUMMARY:smash.:running lib/owncloud/test_remoteShareFile.py in /home/vincent/work/smashdir/test_remoteShareFile as test testset #0 {'test_sharePermissions': 31}
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
SUMMARY:smash.:Elapsed time: 38s (0:00:38.132163)
SUMMARY:smash.:running lib/owncloud/test_remoteShareFile.py in /home/vincent/work/smashdir/test_remoteShareFile as test testset #1 {'test_sharePermissions': 3}
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
2016-03-21 17:31:33,519 - ERROR - sharee_one - shared and shouldn't have error_check(result == -1, "shared and shouldn't have") failed in sharee_one() ["/home/vincent/work/workspace/smashbox/lib/owncloud/test_remoteShareFile.py" at line 197]
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
2016-03-21 17:31:38,566 - ERROR - sharee_two - File /home/vincent/work/smashdir/test_remoteShareFile/sharee_two/TEST_FILE_USER_RESHARE.dat exists but should not error_check(not os.path.exists(fn), "File %s exists but should not" % fn) failed in expect_does_not_exist() ["/home/vincent/work/workspace/smashbox/python/smashbox/utilities/__init__.py" at line 801]
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1
2016-03-21 17:31:42,945 - ERROR - sharee_one - 1 error(s) reported
2016-03-21 17:31:42,952 - ERROR - sharee_two - 1 error(s) reported
SUMMARY:smash.:Elapsed time: 39s (0:00:39.189484)
CRITICAL:smash.:Aborting run -- non-zero exit code (2)
```

nickvergessen committed:
Remote sharing ignores the share permission for now, so sharing should always work:
https://github.com/owncloud/core/issues/22495

nickvergessen commented:
Right, caused by the changes that lead to https://github.com/owncloud/core/issues/22495

Adjusted the test to make sure sharing is always accepted and added FIXME code blocks for the "expected" behaviour

PVince81 commented:
:+1:
